WordPressKit/StatsServiceRemoteV2.swift

```diff
@@ -0,0 +1,193 @@
+        let pathComponent = StatsLastPostInsight.pathComponent
+
+        let path = self.path(forEndpoint: "sites/\(siteID)/\(pathComponent)", withVersion: ._1_1)
+
+        wordPressComRestApi.GET(path, parameters: properties, success: { (response, _) in
+            guard
+                let jsonResponse = response as? [String: AnyObject],
+                let posts = jsonResponse["posts"] as? [[String: AnyObject]],
+                let post = posts.first,
+                let postID = post["ID"] as? Int else {
+                    completion(nil, ResponseError.decodingFailure)
+                    return
+            }
+
+            self.getPostViews(for: postID) { (views, error) in
+                guard
+                    let views = views,
+                    let insight = StatsLastPostInsight(jsonDictionary: post, views: views) else {
+                        completion(nil, ResponseError.decodingFailure)
+                        return
+
+                }
+
+                completion(insight, nil)
+            }
+        }, failure: {(error, _) in
+            completion(nil, error)
+        })
+    }
+
+    public func getPostViews(`for` postID: Int, completion: @escaping ((Int?, Error?) -> Void)) {
+        let parameters = ["fields": "views" as AnyObject]
+
+        let path = self.path(forEndpoint: "sites/\(siteID)/stats/post/\(postID)", withVersion: ._1_1)
+
+        wordPressComRestApi.GET(path,
+                                parameters: parameters,
+                                success: { (response, _) in
+                                    guard
+                                        let jsonResponse = response as? [String: AnyObject],
+                                        let views = jsonResponse["views"] as? Int else {
+                                            completion(nil, ResponseError.decodingFailure)
+                                            return
+                                    }
+                                    completion(views, nil)
+                                }, failure:  { (error, _) in
+                                    completion(nil, error)
+                                }
+        )
+    }
+
+}
+
+// This serves both as a way to get the query properties in a "nice" way,
+// but also as a way to narrow down the generic type in `getInisght:` method.
+public protocol InsightProtocol {
+    static var queryProperties: [String: AnyObject] { get }
+    static var pathComponent: String { get }
+
+    init?(jsonDictionary: [String: AnyObject])
+}
+
+extension InsightProtocol {
+
+    // A big chunk of those use the same endpoint and queryProperties.. Let's simplify the protocol conformance in those cases.
+
+    public static var queryProperties: [String: AnyObject] {
+        return [:]
+    }
+
+    public static var pathComponent: String {
+        return "stats/"
+    }
+}
+
+// For some god-forsaken reason Swift compiler freaks out if this is not declared _in this file_,
```

Comment:
The content of this comment is valid.

From what I can tell, this is the only type you refer to the static members of within this file. That is to say, none of the other eight `InsightProtocol`-conformers are causing problems because we don't attempt to access them. I tested one in a local branch to confirm.

As for the tone of this comment, I would respectfully request that we moderate it slightly. Instead of:
```
For some god-forsaken reason 
```
please consider:
```
At the moment, I am unable to explain why the
```

Reply:
As for the essence of the problem you observed here & rightly noted in the PR description, I tinkered with this a bit, and wanted to report my findings.

I noted that the problematic type (i.e., `StatsLastPostInsight`) is the one you explicitly reference the static members of. I encountered some difficulty, but ultimately, I wanted to share one approach for your consideration:

1. Create new file `StatsLastPostInsight.swift` and extract the struct & protocol conformance to it.
1. I created an extension of `StatsServiceRemoteV2` in `StatsLastPostInsight.swift`, and moved the "problem" methods there (i.e., `getInsight(completion:)`, `.getLastPostInsight(completion:)`, and `getPostViews(for:completion:)`.

Bad idea? Worst idea? 😂

Reply:
I actually meant to redact that comment before posting this PR, I'm embarassed that I forgot to do that. Thanks for pointing this out!